packages/web-ui-registration/src/RegisterForm.tsx

```diff
@@ -0,0 +1,228 @@
+import { useUniqueId } from '@rocket.chat/fuselage-hooks';
+import { FieldGroup, TextInput, Field, PasswordInput, ButtonGroup, Button, TextAreaInput } from '@rocket.chat/fuselage';
+import { Form, ActionLink } from '@rocket.chat/layout';
+import { useSetting, useTranslation } from '@rocket.chat/ui-contexts';
+import type { ReactElement } from 'react';
+import { useForm } from 'react-hook-form';
+import { Trans } from 'react-i18next';
+
+import type { DispatchLoginRouter } from './hooks/useLoginRouter';
+import { useRegisterMethod } from './hooks/useRegisterMethod';
+
+export const LoginRegisterForm = ({ setLoginRoute }: { setLoginRoute: DispatchLoginRouter }): ReactElement => {
+	const t = useTranslation();
+
+	const requireNameForRegister = Boolean(useSetting('Accounts_RequireNameForSignUp'));
+	const requiresPasswordConfirmation = useSetting('Accounts_RequirePasswordConfirmation');
+
+	const manuallyApproveNewUsersRequired = useSetting('Accounts_ManuallyApproveNewUsers');
+
+	const {
+		register,
+		handleSubmit,
+		setError,
+		watch,
+		formState: { errors },
+	} = useForm<{
+		name: string;
+		passwordConfirmation: string;
+		username: string;
+		password: string;
+		email: string;
+		reason: string;
+	}>();
```

Comment:
I think we should keep consistency in the validation mode. Why `onSubmit` here and why `onChange` in Login?